From f27d242e8b619d8ae867627cc342c8dc0bf762cc Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Thu, 13 Feb 2014 01:04:51 +0000
Subject: [PATCH 14/24] Build MinGW client library using gyp-generated Makefile

Make what it's possible to make with gyp for MinGW.

Tools which rely on DIA (dump_syms, ms_symbol_server_converter) cannot be built
for MinGW.

In theory it should be possible to build symupload, because it doesn't depend on
DIA, but it depends on the common_windows_libs, which does have a dependency on
DIA.

v2:
Use configure to find WINDRES

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
---
 Makefile.am  | 7 +++++++
 configure.ac | 1 +
 2 files changed, 8 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index ad62471..7632ba1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -338,6 +338,13 @@ bin_PROGRAMS += \
 endif
 endif
 
+if MINGW_HOST
+# For MinGW, use gyp to generate a Makefile to build Windows client library
+all-local:
+	cd $(top_srcdir)/src ; gyp --no-circular-check --depth=. -f make client/windows/breakpad_client.gyp
+	cd $(top_srcdir)/src ; $(MAKE) CXX="$(CXX)" CXXFLAGS="-DUNICODE -D_UNICODE $(AM_CPPFLAGS) $(CXXFLAGS)" RC="$(WINDRES)"
+endif
+
 ## Tests
 if !DISABLE_PROCESSOR
 check_PROGRAMS += \
diff --git a/configure.ac b/configure.ac
index a2b9287..b2d8e33 100644
--- a/configure.ac
+++ b/configure.ac
@@ -47,6 +47,7 @@ AM_PROG_CC_C_O
 AC_PROG_CPP
 AC_PROG_CXX
 AC_PROG_RANLIB
+AC_CHECK_TOOL(WINDRES, [windres])
 AM_CONDITIONAL(GCC, test "$GCC" = yes)   # let the Makefile know if we're gcc
 
 dnl This must come before all the feature tests below.
-- 
2.1.1

